src/building/count.c

```diff
@@ -374,3 +377,102 @@ int building_count_fort_type_total(figure_type type)
     }
     return total;
 }
+
+int building_count_roads_in_area(int minx, int miny, int maxx, int maxy)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+            
+            if (map_terrain_is(grid_offset, TERRAIN_ROAD)) {
+                total++;
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_highway_in_area(int minx, int miny, int maxx, int maxy)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+
+            if (map_terrain_is(grid_offset, TERRAIN_HIGHWAY)) {
+                total++;
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_plaza_in_area(int minx, int miny, int maxx, int maxy)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+
+            if (map_terrain_is(grid_offset, TERRAIN_ROAD) && map_property_is_plaza_earthquake_or_overgrown_garden(grid_offset)) {
+                total++;
+            }
+        }
+    }
+    return total;
+}
+
+int building_count_gardens_in_area(int minx, int miny, int maxx, int maxy, int overgrown)
+{
+    int total = 0;
+    int grid_offset;
+    for (int y = miny; y < maxy; y++) {
+        for (int x = minx; x < maxx; x++) {
+            grid_offset = map_grid_offset(x, y);
+
+            if (map_terrain_is(grid_offset, TERRAIN_GARDEN)) {
+                int is_overgrown = map_property_is_plaza_earthquake_or_overgrown_garden(grid_offset) != 0;
+                if (is_overgrown == overgrown) {
+                    total++;
+                }
+            }
+        }
+    }
+    return total;
+}
+
+static int min_x;
+static int min_y;
+
+static int init_count_terrain() {
```

Comment:
```suggestion
static void get_min_map_xy(void) {
```

With this small change it's ready to merge (don't forget to change all calls to this function).